CT-1553 sar deadline calculation

misaka opened the pull request:
SAR cases have their external deadlines calculated using calendar days instead
of business days as FOI cases do. As we could only parameterise on the number of
days up until now, this change adds a new kind of deadline calculator to handle
calendar days, and refactors the deadline calculators to accomodate this.

aliuk2012 commented on app/views/cases/new.html.slim:
```diff
@@ -1,10 +1,12 @@
 = form_for @case do |f|
 
+  = f.radio_button_fieldset :requester_type, choices: [:academic_business_charity, :journalist, :member_of_the_public, :offender, :solicitor, :staff_judiciary, :what_do_they_know], inline: false
+
```
You shouldn't need `inline: false`

EdwardAndress replied:
OK I will remove it

aliuk2012 commented on db/structure.sql:
```diff
@@ -370,6 +387,26 @@ ALTER TABLE ONLY cases
+('20161125115930'),
+('20161130164018'),
+('20161209220224'),
+('20170111115617');
```
multiline new lines, the same problem we had with #66 commit this morning

EdwardAndress replied:
argh!  that sneaked in, sorry

EdwardAndress replied:
Actually... shouldn't this be in?  Maybe the problem is that some previous changes were not checked in.

aliuk2012 commented:
Any particular reason why theres loads of other changes in db/structure.sql in terms of 'cases' and categories etc

EdwardAndress commented:
@aliuk2012 regarding db/structure.sql, since the file is generated automatically I am not sure why there may be unexpected changes... @misaka ?

misaka commented:
@EdwardAndress @aliuk2012 Sounds like the changes to `structure.sql` are likely related to the version. Once we get #73 merged let's fix up the `structure.sql` here (dump it again using pg_dump 9.5.5) and rebase the changes and see how far that gets us.

EdwardAndress commented:
I used the web-based method for resolving a merge conflict, as an experiment, without realising it would generate a merge commit.

EdwardAndress commented:
Otherwise, are we all happy with this now?

misaka commented on db/migrate/20170111115617_add_requester_type_to_case.rb:
```diff
@@ -0,0 +1,20 @@
+      'staff_judiciary',
+      'what_do_they_know'
+
+    add_column :cases, :requester_type, :requester_type
```
What are the chances we'll need to create a filter in the front-end based on this column? If so, we may want to consider putting an index on here, depending on how confident we feel that we'll remember to index it later when we need to add the filter.

EdwardAndress replied:
There may be other places where indexing is also missing.  assignment_type for example

misaka commented on spec/features/02_create/case_create_spec.rb:
```diff
@@ -23,6 +24,8 @@
     expect(current_path).to eq new_case_path
     expect(page).to have_content('New case')
+
+    choose user_input.requester_type
```
I don't see where we're testing that this was saved in the case below, are we?

EdwardAndress replied:
I wonder whether that was lost when I tried to resolve a messy MC.  Anyway, I will add it in now.

misaka committed:
CT-1553 add calendar days deadline calculations for SAR cases

SAR cases have their external deadlines calculated using calendar days instead
of business days as FOI cases do. As we could only parameterise on the number of
days up until now, this change adds a new kind of deadline calculator to handle
calendar days, and refactors the deadline calculators to accomodate this.

misaka committed:
CT-1553 add deadline_calculator_class to CorrespondenceType

This replaces time_limit_type with the actual name of the class to use as the
deadline calculator, just to make it a bit clearer what's going on.

misaka committed:
CT-1553 remove GQs from correspondence type seeding
